__tests__/order/order.query.test.js

```diff
@@ -0,0 +1,348 @@
+                lang
+                value
+              }
+            }
+            delivery {
+              sentBy
+              invoiceNumber
+            }
+            paymentMethod
+          }
+        }`
+    })
+
+    const orders = res.data.getAllOrders;
+
+    expect(orders).toBeDefined();
+    expect(orders.length).toBeGreaterThan(0);
+    expect(orders).toBeInstanceOf(Array);
+    expect(orders.find(order => order._id === orderId)).toMatchObject(orderResult);
+  });
+
+  test('should recive order by id', async () => {
+    const res = await client.query({
+      query: gql`
+        query($id: ID!) {
+          getOrderById(id: $id) {
+            ...on Order {
+              status
+              totalPrice {
+                currency
+                value
+              }
+              user {
+                address {
+                  appartment
+                  city
+                  region
+                  street
+                  country
+                  buildingNumber
+                  zipcode
+                }
+                email
+                lastName
+                firstName
+                phoneNumber
+              }
+              items {
+                bottomColor {
+                  lang
+                  value
+                }
+                closure {
+                  lang
+                  value
+                }
+                model {
+                  lang
+                  value
+                }
+                closureColor 
+                size{
+                  widthInCm
+                  weightInKg
+                  heightInCm
+                  volumeInLiters
+                  depthInCm
+                }
+                additions {
+                  lang
+                  value
+                }
+                actualPrice {
+                  currency
+                  value
+                }
+                name {
+                  lang
+                  value
+                }
+                pattern {
+                  lang
+                  value
+                }
+                category {
+                  lang
+                  value
+                }
+                quantity
+                colors {
+                  lang
+                  value
+                }
+                subcategory {
+                  lang
+                  value
+                }
+                bottomMaterial {
+                  lang
+                  value
+                }
+              }
+              delivery {
+                sentBy
+                invoiceNumber
+              }
+              paymentMethod
+            }
+            ...on Error {
+              statusCode
+              message
+            }
+          }
+        }`,
+      variables: {
+        id: orderId,
+      },
+    });
+
+    const order = res.data.getOrderById
+
+    expect(order).toMatchSnapshot()
+    expect(order).toMatchObject(orderResult);
```

Comment:
It is not the best approach to writing tests. I understand that it is much easier to assert the whole object but it makes tests very unreadable. Please, try to assert at least top-level fields separately. For nested values of fields, it is okay to use objects but for root one - no. Thanks

Reply:
fixed